packages/react-native-sdk/src/utils/push/index.ts

```diff
@@ -0,0 +1,169 @@
+// };
+
+const ACCEPT_CALL_ACTION_ID = 'accept';
+const DECLINE_CALL_ACTION_ID = 'decline';
+
+export async function setupCallkeep() {
+  const callkeep = getCallKeepLib();
+  const pushConfig = getPushConfig();
+  const options: Parameters<RNCallKeepType['setup']>[0] = {
+    ios: {
+      appName: pushConfig.ios_appName,
+      supportsVideo: true,
+    },
+    android: {
+      ...pushConfig.android_phoneCallingAccountPermissionTexts,
+      additionalPermissions: [],
+    },
+  };
+  return callkeep.setup(options).then((accepted) => {
+    if (accepted) {
+      callkeep.setAvailable(true);
+    }
+  });
+}
+
+/** Firebase RemoteMessage handler **/
+export async function setupFirebaseHandlerAndroid(client: StreamVideoClient) {
+  if (Platform.OS !== 'android') {
+    return;
+  }
+  const messaging = getFirebaseMessagingLib();
+  const pushConfig = getPushConfig();
+  const firebaseListener = async (
+    message: FirebaseMessagingTypes.RemoteMessage,
+  ) => {
+    /* Example data from firebase
+      "message": {
+          "data": {
+            call_cid: 'audio_room:dcc1638c-e90d-4dcb-bf3b-8fa7767bfbb0',
+            call_display_name: '',
+            created_by_display_name: 'tommaso',
+            created_by_id: 'tommaso-03dcddb7-e9e2-42ec-b2f3-5043aac666ee',
+            receiver_id: 'martin-21824f17-319b-401b-a61b-fcab646f0d3f',
+            sender: 'stream.video',
+            type: 'call.live_started',
+            version: 'v2'
+          },
+          // other stuff
+      }
+    */
+    const data = message.data;
+    if (!data) {
+      return;
+    }
+    // Check if the message is for Stream Video Call
+    if (data.sender !== 'stream.video') {
+      return;
+    }
+    const { getTitle, getBody } =
+      pushConfig.android_incomingCallNotificationTextGetters;
+    const channelId = pushConfig.android_incomingCallChannel.id;
+    const createdUserName = data.created_by_display_name;
+    await notifee.displayNotification({
+      title: getTitle(createdUserName),
+      body: getBody(createdUserName),
+      data,
+      android: {
+        channelId,
+        ongoing: true,
+        pressAction: {
+          id: 'default',
+          launchActivity: 'default',
+        },
+        actions: [
+          {
+            title: 'Accept',
+            pressAction: {
+              id: ACCEPT_CALL_ACTION_ID,
+              launchActivity: 'default',
+            },
+          },
+          {
+            title: 'Decline',
+            pressAction: {
+              id: DECLINE_CALL_ACTION_ID,
+              launchActivity: 'default',
+            },
+          },
+        ],
+      },
+    });
+    // const callkeep = getCallKeepLib();
+    // const uuid = message.data.call_cid;
+    // const localizedCallerName = message.data.created_by_display_name;
+    // const handle = 'handle'; // Phone number of the caller // TODO: unclear what this is used for?!
+    // callkeep.displayIncomingCall(uuid, handle, localizedCallerName);
```

Comment:
ignore above comment, we may swap to callkeep instead of notifee for incoming call in android